test/Test/Cardano/Prelude/Tripping.hs

```diff
@@ -36,11 +71,62 @@ roundTripsAesonShow
   :: (Eq a, MonadTest m, ToJSON a, FromJSON a, Show a) => a -> m ()
 roundTripsAesonShow a = tripping a encode decode
 
--- | Round trip any `a` with both `ToJSON` and `FromJSON` instances
+-- | Round trip any `a` with both `ToJSON` and `FromJSON` instances.
 roundTripsAesonBuildable
   :: (Eq a, MonadTest m, ToJSON a, FromJSON a, Buildable a) => a -> m ()
 roundTripsAesonBuildable a = trippingBuildable a encode decode
 
+-- | Round trip any `a` with both `ToJSON` and `FromJSON` canonical instances.
+roundTripsCanonicalJson
+  :: ( Eq a
+     , Show a
+     , MonadTest m
+     , CanonicalJSON.ToJSON Identity a
+     , CanonicalJSON.FromJSON (Either SchemaError) a
+     )
+  => a
+  -> m ()
+roundTripsCanonicalJson a = tripping a canonicalEncode canonicalDecode
+ where
+  canonicalEncode
+    :: forall a . CanonicalJSON.ToJSON Identity a => a -> LB.ByteString
+  canonicalEncode x =
+    CanonicalJSON.renderCanonicalJSON $ runIdentity $ CanonicalJSON.toJSON x
+  canonicalDecode
+    :: forall a . CanonicalJSON.FromJSON (Either SchemaError) a
+    => LB.ByteString
+    -> Either SchemaError a
+  canonicalDecode y =
+    CanonicalJSON.fromJSON
+      $ either (panic . toS) identity
+      $ CanonicalJSON.parseCanonicalJSON y
+
+-- | Pretty round trip any `a` with both `ToJSON` and `FromJSON` canonical instances.
+roundTripsCanonicalJsonPretty
+  :: ( Eq a
+     , Show a
+     , MonadTest m
+     , CanonicalJSON.ToJSON Identity a
+     , CanonicalJSON.FromJSON (Either SchemaError) a
+     )
+  => a
+  -> m ()
+roundTripsCanonicalJsonPretty a = tripping a canonicalEncPre canonicalDecPre
+ where
+  canonicalEncPre
+    :: forall a . CanonicalJSON.ToJSON Identity a => a -> LB.ByteString
+  canonicalEncPre x =
+      LB.fromStrict . encodeUtf8 . toS $
+      CanonicalJSON.prettyCanonicalJSON $ runIdentity $ CanonicalJSON.toJSON x
+  canonicalDecPre
```

Comment:
Don't need this definition twice if the pretty version doesn't do anything different